core/src/main/java/com/novoda/merlin/receiver/ConnectivityChangesRegister.java

```diff
@@ -15,38 +15,35 @@
     private final Context context;
     private final ConnectivityManager connectivityManager;
     private final AndroidVersion androidVersion;
-    private final MerlinService merlinService;
 
     private ConnectivityReceiver connectivityReceiver;
     private ConnectivityCallbacks connectivityCallbacks;
 
     public ConnectivityChangesRegister(Context context,
                                        ConnectivityManager connectivityManager,
-                                       AndroidVersion androidVersion,
-                                       MerlinService merlinService) {
+                                       AndroidVersion androidVersion) {
         this.context = context;
         this.connectivityManager = connectivityManager;
         this.androidVersion = androidVersion;
-        this.merlinService = merlinService;
     }
 
-    public void register() {
+    public void register(MerlinService.ConnectivityChangesListener connectivityChangesListener) {
```

Comment:
`ConnectivityChangesRegister` is created in the `MerlinServiceBinder` but the callbacks need to be exposed in the `MerlinService` so we pass through the method call instead.